File: dotnet-service/src/Enums/SourceType.cs
namespace dotnet_service.Enums;

public enum SourceType
{
    DotnetService = 1,
    SpringBootService = 2
}

public static class SourceTypeUtils
{
    public static string ToName(this SourceType type)
    {
        return type switch
        {
            SourceType.DotnetService => "DotnetService",
            SourceType.SpringBootService => "SpringBootService",
            _ => ""
        };
    }
    
    public static SourceType FromName(string name)
    {
        return name.ToLower() switch
        {
            "dotnetservice" => SourceType.DotnetService,
            "springbootservice" => SourceType.SpringBootService,
            _ => SourceType.DotnetService
        };
    }
}

File: springboot-service/src/main/java/com/service/app/enums/SourceType.java
package com.service.app.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum SourceType {
    DOTNET_SERVICE("DotnetService"),
    SPRINGBOOT_SERVICE("SpringBootService");

    private final String name;

    public static SourceType fromName(String name){
        if(name.equalsIgnoreCase("dotnetservice")) return DOTNET_SERVICE;
        return SPRINGBOOT_SERVICE;
    }
}
